openedx/core/djangoapps/course_date_signals/handlers.py

```diff
@@ -78,12 +80,46 @@ def _gather_graded_items(root, due):  # lint-amnesty, pylint: disable=missing-fu
     return []
 
 
+def _get_custom_pacing_children(subsection):
+    """
+    Return relative date items for the subsection and its children
+    """
+    items = [subsection]
+    section_date_items = []
+    while items:
+        next_item = items.pop()
+        num_weeks = next_item.fields['due_num_weeks'].read_from(next_item)
+        if num_weeks:
+            section_date_items.append((next_item.location, {'due': timedelta(weeks=num_weeks)}))
+        items.extend(next_item.get_children())
+    return section_date_items
+
+
 def extract_dates_from_course(course):
     """
     Extract all dates from the supplied course.
     """
     log.info('Extracting course dates for %s', course.id)
-    if course.self_paced:
+    if course.self_paced and custom_pls_is_active(course.id):
+        metadata = _field_values(FIELDS_TO_EXTRACT, course)
+        # self-paced courses may accidentally have a course due date
+        metadata.pop('due', None)
+        date_items = [(course.location, metadata)]
+        if SelfPacedRelativeDatesConfig.current(course_key=course.id).enabled:
+            # If custom pacing is set on a subsection, apply the set relative
+            # date to all the content inside the subsection. Otherwise
+            # apply the current PLS logic for self paced courses.
+            for _, section, weeks_to_complete in spaced_out_sections(course):
+                section_date_items = []
+                for subsection in section.get_children():
+                    if subsection.fields['due_num_weeks'].read_from(subsection):
+                        section_date_items.extend(_get_custom_pacing_children(subsection))
```

Comment:
Also just realized, right now, `_get_custom_pacing_children` is relying on all children blocks of the subsection to have the `due_num_weeks` field defined, but I believe you could just pass in that value and set it for all of the children. That is what we do for the `_gather_graded_items` function where it's the same due date for all children within the section. (We are technically allowing even more granularity by doing it on a subsection vs section basis already)

Reply:
Ohh sounds good, and I also realized I should put a condition in `_get_custom_pacing_children` where I don't assign due dates to ORA content since it has its own due date when created.